beta/iam/guides/validate-messages-in-griffon.md

```diff
@@ -0,0 +1,129 @@
+
+1. In the event list, select the event with type **AEP Response Event Handle**. There will likely be several events with this type - ensure the one selected has an **AEPExtensionEventSource** of `personalization:decisions`
+
+    ![AEP Response Event Handle](./../../../.gitbook/assets/message_response.png)
+
+1. Expand the **Payload** section in the right window, and drill-down the tree until you find the **items** array. The full path to **items** is:
+
+    ```
+    ACPExtensionEventData.payload.0.items
+    ```
+
+    This array contains an entry for each of this app's published messages. As shown in the screenshot below, the message is fully defined in the **content** property of each item's **data** object:
+
+    ![AEP Response Event Handle Payload](./../../../.gitbook/assets/message_response_payload.png)
+
+## Use the In-App Messaging Assurance UI plugin
+
+Once all of the above validation sections are complete, you can use the **In-App Messaging** plugin view in the Assurance UI to further debug your app.
+
+#### Install the In-App Messaging plugin
+
+> If you have already installed the **In-App Messaging** plugin in your Assurance UI setup, skip this section.
+
+1. In the Assurance UI, click on **Configure** button at the bottom of the left-rail navigation
+
+1. Search for the row named **In-App Messaging** under the **ADOBE JOURNEY OPTIMIZER (BETA)** heading, and click the **+** button on its right
+
+1. Click the **Save** button
+
+    ![Install the In-App Messaging plugin](./../../../.gitbook/assets/install_iam_plugin.png)
+
+#### Inspecting a downloaded message
+
+Using the IAM plugin you can do the following for each message downloaded by the client:
+
+- In the **Rules** tab - view the rules defining when the message will be shown to the user
+- In the **History** tab - review a history of client events, including a comparison between the event's contents and the message's triggering criteria
+- In the **Message Preview** window - see a preview of the message's html
+- In the **Message Behavior** window - review message behavior, including its supported gestures and animations
+- In the **Message Behavior** window - review message size and positioning properties
+- Clicking the **Simulate on Device** button - trigger the currently selected message, causing it to be displayed on the connected client
+
+    ![Inspecting a downloaded message](./../../../.gitbook/assets/iam_simulation.png)
+
+## FAQs
+
+**Q:** What do I do when one of the required extensions is missing? <br />
+**A:** Ensure that each required extension is linked to your project and registered by `MobileCore`. Ref: [registerExtensions](./../../../foundation-extensions/mobile-core/mobile-core-api-reference.md#registerextensions)
+
+**Q:** Why can't I find an event named `Edge Optimize Personalization Request`? <br />
+**A:** Check to ensure each required extensions is on a version that meets its minimum requirement.
```

Comment:
"Ensure that each of the required extensions is at the minimum required version."